### DISCUSSION TITLE
Is tht.dev created with tht?

### DISCUSSION THREAD

**Discussion by aoirthoir** [joelesko/tht#21] [General]
Is the tht.dev website created with tht?

Also would you be willing to upload the tht website to this github, especially if it is built with tht? But even if it is not, so that way I could snap the pieces out easier to put all the pages combined together into one big page and then into a properly formed PDF manual?

Obviously I am interested in the <main> sections and i could run through and remove the CSS if necessary.

Right now I am just combining them with the setup now from a download of the site..which means certain sections that werent meant to be combined, when i combine them give funny looks. 

if you want to message me i can give you the testing link I have it at right now. 

Thanks

**Comment by joelesko** [comment]
Yes, tht.dev is built with THT.  I will probably post the code if it gets to a point where it needs community updates. 

I'm not sure for your purposes how it would help, since it's mostly dynamically generated.

Sure, you can send me a link and I can take a look.

**Comment by aoirthoir** [comment]
I presum each page's "content" is wrapped in a full html page via tht templates. If those sections are separate, then i could combine them in a page with other content... such as all methods from /manual/class/list. 

I find it easier to keep paging down reading all list methods for instance rather than clicking to one list method, clicking back, then clicking to the next then clicking back etc. And of course for a PDF manual, having access to those up to date bite size content chunks means i can just run a script to combine them.

As of now what I did was downoad the entire site with HTTrack, which gave me some weird quirks in output. 

Then I went and combined pages in groups based on their underlings (manual/class/list/everything into /manual/class/list.html etc). 

Then I used print preview on firefox... because all the HTML to PDF software I have tried has been dissatisfying.

Then whenever a new h1/h2/h3 (based on my reformatting of the html) was hit,  i create a new page in css with style="page-break-before: always;" in the h tag. 

If a heading/subheading fit on a page well enough, if not then I repeated it after checking paging. So that any time im on a page i could instantly tell what was going on...

This also meant not splitting up code examples if it could be avoided.. sometimes removing blank lines.

Also I tried to put everything in as much order as possible as the website. 

So I think having the tht.dev on github would mean that I could clone (or fork?) and then with some scripts do most of that automatically... though this time around I did it almost entirely by hand (with some vs code search and replace).
